src/Controller/Api/Legacy/GroupController.php

```diff
@@ -0,0 +1,275 @@
+    {
+        $this->groupManager = $groupManager;
+        $this->formFactory = $formFactory;
+    }
+
+    /**
+     * Returns a paginated list of groups.
+     *
+     * @ApiDoc(
+     *  resource=true,
+     *  output={"class"="Sonata\DatagridBundle\Pager\PagerInterface", "groups"={"sonata_api_read"}}
+     * )
+     *
+     * @Get("/groups")
+     *
+     * @QueryParam(name="page", requirements="\d+", default="1", description="Page for groups list pagination (1-indexed)")
+     * @QueryParam(name="count", requirements="\d+", default="10", description="Number of groups by page")
+     * @QueryParam(name="orderBy", map=true, requirements="ASC|DESC", nullable=true, strict=true, description="Query groups order by clause (key is field, value is direction")
+     * @QueryParam(name="enabled", requirements="0|1", nullable=true, strict=true, description="Enabled/disabled groups only?")
+     *
+     * @View(serializerGroups={"sonata_api_read"}, serializerEnableMaxDepthChecks=true)
+     *
+     * @return PagerInterface
+     */
+    public function getGroupsAction(ParamFetcherInterface $paramFetcher)
+    {
+        $supportedFilters = [
+            'enabled' => '',
+        ];
+
+        $page = $paramFetcher->get('page');
+        $limit = $paramFetcher->get('count');
+        $sort = $paramFetcher->get('orderBy');
+        $criteria = array_intersect_key($paramFetcher->all(), $supportedFilters);
+
+        foreach ($criteria as $key => $value) {
+            if (null === $value) {
+                unset($criteria[$key]);
+            }
+        }
+
+        if (!$sort) {
+            $sort = [];
+        } elseif (!\is_array($sort)) {
+            $sort = [$sort, 'asc'];
+        }
+
+        return $this->groupManager->getPager($criteria, $page, $limit, $sort);
+    }
+
+    /**
+     * Retrieves a specific group.
+     *
+     * @ApiDoc(
+     *  requirements={
+     *      {"name"="id", "dataType"="integer", "requirement"="\d+", "description"="group id"}
+     *  },
+     *  output={"class"="FOS\UserBundle\Model\GroupInterface", "groups"={"sonata_api_read"}},
+     *  statusCodes={
+     *      200="Returned when successful",
+     *      404="Returned when group is not found"
+     *  }
+     * )
+     *
+     * @Get("/group/{id}")
+     *
+     * @View(serializerGroups={"sonata_api_read"}, serializerEnableMaxDepthChecks=true)
+     *
+     * @param $id
+     *
+     * @return GroupInterface
+     */
+    public function getGroupAction($id)
+    {
+        return $this->getGroup($id);
+    }
+
+    /**
+     * Adds a group.
+     *
+     * @ApiDoc(
+     *  input={"class"="sonata_user_api_form_group", "name"="", "groups"={"sonata_api_write"}},
+     *  output={"class"="Sonata\UserBundle\Model\Group", "groups"={"sonata_api_read"}},
+     *  statusCodes={
+     *      200="Returned when successful",
+     *      400="Returned when an error has occurred while group creation",
+     *  }
+     * )
+     *
+     * @Post("/group")
```

Comment:
```suggestion
     * @REST\Post("/group")
```
The same here for `@REST`.

Reply:
Done.